nima/http/media/media/src/main/java/io/helidon/nima/http/media/ReadableEntityBase.java

```diff
@@ -108,13 +126,20 @@ public void consume() {
         if (consumed()) {
             return;
         }
+        if (!entityRequested.getAndSet(true)) {
+            entityRequestedCallback.accept(true);
+        }
         InputStream theStream;
         if (inputStream == null) {
             theStream = inputStream();
         } else {
             theStream = inputStream;
         }
         try (theStream) {
+            if (theStream.available() == 0) {
+                entityProcessedRunnable.run();
+                return;
+            }
```

Comment:
@tomas-langer FYI not sure if its correct approach, otherwise: calling non-existent route with `expects 100` client doesn't ever send data as there was no `100-continue` but server get stuck on the stream `is.read(` when draining/consuming entity before 404 error handling

Reply:
Never mind :blush: